### PR TITLE
"has tags" filter

### PR DESCRIPTION
**Pull request by lelemm**
New filter to make it simple to filter tags. Instead of regex directly, encapsulate the regex into a "has tag" filter. This filter should be only available to "notes" field.

### CODE REVIEW

**Comment by lelemm** [comment]
https://github.com/user-attachments/assets/8d93513d-cdfa-424a-82f9-0b7d8da4aca5

https://discord.com/channels/937901803608096828/1273371290392334386

File: upcoming-release-notes/3290.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+category: Enhancements
+authors: [MatissJanis]
```
Comment (youngcw) [inline upcoming-release-notes/3290.md:3]: forgot to change this it seems

File: packages/desktop-client/src/components/filters/FilterExpression.tsx [hunk 2]
```diff
@@ -79,7 +79,8 @@ export function FilterExpression<T extends RuleConditionEntity>({
                 valueIsRaw={
                   op === 'contains' ||
                   op === 'matches' ||
-                  op === 'doesNotContain'
+                  op === 'doesNotContain' ||
+                  op === 'tags'
```
Comment (jfdoming) [inline packages/desktop-client/src/components/filters/FilterExpression.tsx:83]: nit: what are your thoughts on calling this `hasTags` instead? Thinking if we want to add a `doesNotHaveTags` in the future

File: upcoming-release-notes/3290.md (hunk 1 above)
Comment (lelemm) [reply upcoming-release-notes/3290.md]: ops :)

File: packages/desktop-client/src/components/filters/FilterExpression.tsx (hunk 2 above)
Comment (lelemm) [reply packages/desktop-client/src/components/filters/FilterExpression.tsx]: adopted your naming suggestion

**Comment by lelemm** [comment]
@youngcw @jfdoming mind reviewing again?

**Comment by youngcw** [comment]
This would be good for later, but it would be nice to have a way to maybe have an "or" list of tags instead of just an "and" list.

**Review by youngcw — APPROVED** [review]

**Pull request merged**